model/ResultsService.php

```diff
@@ -848,29 +848,45 @@ public function getVariableDataFromDeliveryResult($resultIdentifier, $wantedType
         return $this->extractTestVariables($this->getVariablesFromObjectResult($testCallIds), $wantedTypes);
     }
 
-    public function extractTestVariables($variableObjects, $wantedTypes)
+    public function extractTestVariables(array $variableObjects, array $wantedTypes, string $filter = self::VARIABLES_FILTER_ALL)
     {
-        $variablesData = [];
-        foreach ($variableObjects as $variable) {
-            if (is_null($variable[0]->callIdItem) && in_array(get_class($variable[0]->variable), $wantedTypes)) {
-                $variablesData[] = $variable[0]->variable;
-            }
-        }
-        usort($variablesData, function ($a, $b) {
-            list($usec, $sec) = explode(" ", $a->getEpoch());
-            $floata = ((float) $usec + (float) $sec);
-            list($usec, $sec) = explode(" ", $b->getEpoch());
-            $floatb = ((float) $usec + (float) $sec);
+        $variableObjects = array_filter($variableObjects, static function (array $variableObject) use ($wantedTypes) {
+            $variable = current($variableObject);
 
-            if ((floatval($floata) - floatval($floatb)) > 0) {
-                return 1;
-            } elseif ((floatval($floata) - floatval($floatb)) < 0) {
-                return -1;
-            } else {
-                return 0;
+            return $variable->callIdItem === null && in_array(get_class($variable->variable), $wantedTypes, true);
+        });
+
+        $variableObjects = array_map(static function (array $variableObject) {
+            return current($variableObject)->variable;
+        }, $variableObjects);
+
+        usort($variableObjects, static function (
+            taoResultServer_models_classes_Variable $a,
+            taoResultServer_models_classes_Variable $b
+        ) use ($filter) {
+            if ($filter === self::VARIABLES_FILTER_LAST_SUBMITTED) {
+                return tao_helpers_Date::getTimeStamp($b->getEpoch()) - tao_helpers_Date::getTimeStamp($a->getEpoch());
             }
+
+            return tao_helpers_Date::getTimeStamp($a->getEpoch()) - tao_helpers_Date::getTimeStamp($b->getEpoch());
```

Comment:
please use $a->getCreationTime() instead of tao_helpers_Date::getTimeStamp($a->getEpoch())

Reply:
thanks, fixed